tests/models/test_model.py

```diff
@@ -1930,3 +1933,99 @@ def test_head_dim_8_triton_mqa_attn(batch_size: int = 2):
         output = model(batch)
 
     assert not torch.isnan(output.logits).any()
+
+@pytest.mark.world_size(2)
+@pytest.mark.gpu
+def test_tp_qkvo():
```

Comment:
This test checks that the model weights are TP'd against taking a model and TP-ing its weights with the same TP-ing tool (which itself is NOT tested).
Can we create an un-TP tool and test un-TP-ing the model against the full model?

Can we add an end to end test?
ie fwd pass a sample through both models and verify that outputs are the same; then bwd to compute gradients and un-TP grads to verify they are the same?